src/components/Hero/HeroInfobox.tsx

```diff
@@ -0,0 +1,219 @@
+import {
+  type HeroSideProps,
+  type HeroButtonProps,
+  type HeroHeadingProps,
+  type HeroFloatingProps,
+} from "./Hero"
+import { HeroDropdown, type HeroDropdownProps } from "./HeroDropdown"
+
+type HeroInfoboxVariants =
+  | HeroSideProps["variant"]
+  | HeroFloatingProps["variant"]
+
+export interface HeroInfoboxProps extends HeroHeadingProps, HeroButtonProps {
+  alignment?: "left" | "right"
+  backgroundColor?: "black" | "white" | "gray"
+  size?: "sm" | "md"
+  dropdown?: HeroDropdownProps
+}
+
+const BP_BUTTON_CLASSES =
+  "rounded-none box-content appearance-none items-center border border-solid border-[#f0f0f0] shadow-none inline-flex text-base h-9 justify-center px-3 py-[calc(0.375rem-1px)] relative align-top select-none cursor-pointer text-center whitespace-nowrap focus:outline-none active:outline-none disabled:cursor-not-allowed"
+
+const bgColor: {
+  [key in NonNullable<HeroInfoboxProps["backgroundColor"]>]: string
+} = {
+  black: "bg-canvas-inverse",
+  white: "bg-canvas-base",
+  gray: "bg-canvas-translucentGrey",
+}
+
+const textColor: {
+  [key in NonNullable<HeroInfoboxProps["backgroundColor"]>]: string
+} = {
+  black: "text-content-inverse",
+  white: "text-content-base",
+  gray: "text-content-inverse",
+}
+
+const width: {
+  [key in HeroInfoboxVariants]: {
+    [key in NonNullable<HeroInfoboxProps["size"]>]: string
+  }
+} = {
+  side: {
+    sm: "w-min w-1/3 xl:w-1/2",
+    md: "w-1/2",
+  },
+  floating: {
+    sm: "w-1/3",
+    md: "w-1/2",
+  },
+}
+
+export const HeroInfoboxDesktop = ({
```

Comment:
is it possible to simplify the code structure by operating under some invariants? (whether enforced or not)
e.g
`default -> HeroInfoBoxMobile`
`md -> HeroInfoBoxTablet`
`lg onwards -> HeroInfoBoxDesktop`

asking because I feel like there seems to be quite a lot of responsive css-ing going on in those 3 components, which affects readability and ability to reason about our UI
Plus its confusing why `HeroInfoBoxTablet` and `HeroInfoBarMobile` would have `lg:xxx` classes

I think would be neater to stick to those invariants, then we just need to make sure we use the usual `md:hidden` pattern when consuming the 3 components like you've done in `HeroSide` and `HeroFloating`



edit: I just realised that `HeroSide` only uses Mobile and Desktop, but I think this still makes sense, just that maybe need to make Desktop compatible with both `md` and `lg` 🤔